SingularityUI/app/components/requestForm/RequestForm.jsx

```diff
@@ -0,0 +1,563 @@
+                generateSelectBox: true,
+                selectBoxOptions: {containerCssClass : "select2-select-box select-box-small"}
+              }}
+            />
+          </div>
+        </div>
+      </div>
+    );
+    const numRetriesOnFailure = (
+      <div className='form-group'>
+        <label htmlFor='retries-on-failure'>Number of retries on failure</label>
+        <FormField
+          id='retries-on-failure'
+          prop={{
+            updateFn: event => this.updateField('numRetriesOnFailure', event.target.value),
+            inputType: 'text',
+            value: this.getValue('numRetriesOnFailure')
+          }}
+        />
+      </div>
+    );
+    const killOldNonLongRunningTasksAfterMillis = (
+      <div className='form-group'>
+        <label htmlFor='killOldNRL'>Kill cleaning task(s) after</label>
+        <div className="input-group">
+          <FormField
+            id='killOldNRL'
+            prop={{
+              updateFn: event => this.updateField('killOldNonLongRunningTasksAfterMillis', event.target.value),
+              inputType: 'text',
+              value: this.getValue('killOldNonLongRunningTasksAfterMillis')
+            }}
+          />
+          <div className="input-group-addon">milliseconds</div>
+        </div>
+      </div>
+    );
+    const scheduledExpectedRuntimeMillis = (
+      <div className='form-group'>
+        <label htmlFor='expected-runtime'>Maximum task duration</label>
+        <div className="input-group">
+          <FormField
+            id='expected-runtime'
+            prop={{
+              updateFn: event => this.updateField('scheduledExpectedRuntimeMillis', event.target.value),
+              inputType: 'text',
+              value: this.getValue('scheduledExpectedRuntimeMillis')
+            }}
+          />
+          <div className="input-group-addon">milliseconds</div>
+        </div>
+      </div>
+    );
+    return (
+      <div>
+        { this.shouldRenderField('instances') ? instances : null }
+        { this.shouldRenderField('rackSensitive') ? rackSensitive : null }
+        { this.shouldRenderField('hideEvenNumberAcrossRacksHint') ? hideEvenNumberAcrossRacksHint : null }
+        { this.shouldRenderField('loadBalanced') ? this.renderLoadBalanced() : null }
+        { this.shouldRenderField('waitAtLeastMillisAfterTaskFinishesForReschedule') ? waitAtLeastMillisAfterTaskFinishesForReschedule : null }
+        { this.shouldRenderField('rackAffinity') ? rackAffinity : null }
+        { this.shouldRenderField(CRON_SCHEDULE) || this.shouldRenderField(QUARTZ_SCHEDULE) ? schedule : null }
+        { this.shouldRenderField('numRetriesOnFailure') ? numRetriesOnFailure : null }
+        { this.shouldRenderField('killOldNonLongRunningTasksAfterMillis') ? killOldNonLongRunningTasksAfterMillis : null }
+        { this.shouldRenderField('scheduledExpectedRuntimeMillis') ? scheduledExpectedRuntimeMillis : null }
```

Comment:
minor nitpick but `condition && component` is a pattern i've seen that's a little more succinct than using the ternary operator